diff --git a/node_modules/@aries-framework/core/build/modules/routing/RoutingEvents.d.ts b/node_modules/@aries-framework/core/build/modules/routing/RoutingEvents.d.ts
index 578a26f..3cb7221 100644
--- a/node_modules/@aries-framework/core/build/modules/routing/RoutingEvents.d.ts
+++ b/node_modules/@aries-framework/core/build/modules/routing/RoutingEvents.d.ts
@@ -2,11 +2,12 @@ import type { KeylistUpdate } from './messages/KeylistUpdateMessage';
 import type { MediationState } from './models/MediationState';
 import type { MediationRecord } from './repository/MediationRecord';
 import type { BaseEvent } from '../../agent/Events';
-import type { Routing } from '../connections';
+import type { ConnectionRecord, Routing } from '../connections';
 export declare enum RoutingEventTypes {
     MediationStateChanged = "MediationStateChanged",
     RecipientKeylistUpdated = "RecipientKeylistUpdated",
-    RoutingCreatedEvent = "RoutingCreatedEvent"
+    RoutingCreatedEvent = "RoutingCreatedEvent",
+    ForwardMessageEvent = "ForwardMessageEvent"
 }
 export interface RoutingCreatedEvent extends BaseEvent {
     type: typeof RoutingEventTypes.RoutingCreatedEvent;
@@ -28,3 +29,9 @@ export interface KeylistUpdatedEvent extends BaseEvent {
         keylist: KeylistUpdate[];
     };
 }
+export interface ForwardMessageEvent extends BaseEvent {
+    type: typeof RoutingEventTypes.ForwardMessageEvent;
+    payload: {
+        connectionRecord: ConnectionRecord;
+    };
+}
diff --git a/node_modules/@aries-framework/core/build/modules/routing/RoutingEvents.js b/node_modules/@aries-framework/core/build/modules/routing/RoutingEvents.js
index ad9e68b..1aa5cb6 100644
--- a/node_modules/@aries-framework/core/build/modules/routing/RoutingEvents.js
+++ b/node_modules/@aries-framework/core/build/modules/routing/RoutingEvents.js
@@ -6,5 +6,6 @@ var RoutingEventTypes;
     RoutingEventTypes["MediationStateChanged"] = "MediationStateChanged";
     RoutingEventTypes["RecipientKeylistUpdated"] = "RecipientKeylistUpdated";
     RoutingEventTypes["RoutingCreatedEvent"] = "RoutingCreatedEvent";
+    RoutingEventTypes["ForwardMessageEvent"] = "ForwardMessageEvent";
 })(RoutingEventTypes = exports.RoutingEventTypes || (exports.RoutingEventTypes = {}));
 //# sourceMappingURL=RoutingEvents.js.map
\ No newline at end of file
diff --git a/node_modules/@aries-framework/core/build/modules/routing/handlers/ForwardHandler.js b/node_modules/@aries-framework/core/build/modules/routing/handlers/ForwardHandler.js
index bb61eee..8e2069b 100644
--- a/node_modules/@aries-framework/core/build/modules/routing/handlers/ForwardHandler.js
+++ b/node_modules/@aries-framework/core/build/modules/routing/handlers/ForwardHandler.js
@@ -12,6 +12,8 @@ class ForwardHandler {
     async handle(messageContext) {
         const { encryptedMessage, mediationRecord } = await this.mediatorService.processForwardMessage(messageContext);
         const connectionRecord = await this.connectionService.getById(messageContext.agentContext, mediationRecord.connectionId);
+        // Send forward message to the event emitter so that it can be picked up events
+        await this.mediatorService.emitForwardEvent(messageContext.agentContext, connectionRecord);
         // The message inside the forward message is packed so we just send the packed
         // message to the connection associated with it
         await this.messageSender.sendPackage(messageContext.agentContext, {
diff --git a/node_modules/@aries-framework/core/build/modules/routing/services/MediatorService.d.ts b/node_modules/@aries-framework/core/build/modules/routing/services/MediatorService.d.ts
index e02dbab..ec9ba0d 100644
--- a/node_modules/@aries-framework/core/build/modules/routing/services/MediatorService.d.ts
+++ b/node_modules/@aries-framework/core/build/modules/routing/services/MediatorService.d.ts
@@ -2,6 +2,7 @@ import type { AgentContext } from '../../../agent';
 import type { InboundMessageContext } from '../../../agent/models/InboundMessageContext';
 import type { Query } from '../../../storage/StorageService';
 import type { EncryptedMessage } from '../../../types';
+import type { ConnectionRecord } from '../../connections';
 import type { ForwardMessage, MediationRequestMessage } from '../messages';
 import { EventEmitter } from '../../../agent/EventEmitter';
 import { Logger } from '../../../logger';
@@ -23,6 +24,7 @@ export declare class MediatorService {
         mediationRecord: MediationRecord;
         encryptedMessage: EncryptedMessage;
     }>;
+    emitForwardEvent(agentContext: AgentContext, connectionRecord: ConnectionRecord): Promise<void>;
     processKeylistUpdateRequest(messageContext: InboundMessageContext<KeylistUpdateMessage>): Promise<KeylistUpdateResponseMessage>;
     createGrantMediationMessage(agentContext: AgentContext, mediationRecord: MediationRecord): Promise<{
         mediationRecord: MediationRecord;
diff --git a/node_modules/@aries-framework/core/build/modules/routing/services/MediatorService.js b/node_modules/@aries-framework/core/build/modules/routing/services/MediatorService.js
index ec8cd1b..2118939 100644
--- a/node_modules/@aries-framework/core/build/modules/routing/services/MediatorService.js
+++ b/node_modules/@aries-framework/core/build/modules/routing/services/MediatorService.js
@@ -61,6 +61,14 @@ let MediatorService = class MediatorService {
             mediationRecord,
         };
     }
+    async emitForwardEvent(agentContext, connectionRecord) {
+        this.eventEmitter.emit(agentContext, {
+            type: RoutingEvents_1.RoutingEventTypes.ForwardMessageEvent,
+            payload: {
+                connectionRecord,
+            },
+        });
+    }
     async processKeylistUpdateRequest(messageContext) {
         // Assert Ready connection
         const connection = messageContext.assertReadyConnection();
