diff --git a/node_modules/@credo-ts/core/build/agent/MessageSender.js b/node_modules/@credo-ts/core/build/agent/MessageSender.js
index 689b024..656a3b5 100644
--- a/node_modules/@credo-ts/core/build/agent/MessageSender.js
+++ b/node_modules/@credo-ts/core/build/agent/MessageSender.js
@@ -75,7 +75,7 @@ let MessageSender = class MessageSender {
         this.logger.debug('Sending message');
         await session.send(agentContext, encryptedMessage);
     }
-    async sendPackage(agentContext, { connection, encryptedMessage, recipientKey, options, }) {
+    async sendPackage(agentContext, { connection, encryptedMessage, recipientKey, options, messageType, }) {
         var _a, e_1, _b, _c;
         var _d;
         const errors = [];
@@ -145,6 +145,7 @@ let MessageSender = class MessageSender {
                 connectionId: connection.id,
                 recipientDids: [(0, helpers_1.verkeyToDidKey)(recipientKey)],
                 payload: encryptedMessage,
+                messageType,
             });
             return;
         }
diff --git a/node_modules/@credo-ts/core/build/modules/message-pickup/storage/MessagePickupRepositoryOptions.d.ts b/node_modules/@credo-ts/core/build/modules/message-pickup/storage/MessagePickupRepositoryOptions.d.ts
index decb61a..b0c4342 100644
--- a/node_modules/@credo-ts/core/build/modules/message-pickup/storage/MessagePickupRepositoryOptions.d.ts
+++ b/node_modules/@credo-ts/core/build/modules/message-pickup/storage/MessagePickupRepositoryOptions.d.ts
@@ -13,6 +13,7 @@ export interface AddMessageOptions {
     connectionId: string;
     recipientDids: string[];
     payload: EncryptedMessage;
+    messageType?: string;
 }
 export interface RemoveMessagesOptions {
     connectionId: string;
diff --git a/node_modules/@credo-ts/core/build/modules/routing/services/MediatorService.js b/node_modules/@credo-ts/core/build/modules/routing/services/MediatorService.js
index 6818a3b..6474136 100644
--- a/node_modules/@credo-ts/core/build/modules/routing/services/MediatorService.js
+++ b/node_modules/@credo-ts/core/build/modules/routing/services/MediatorService.js
@@ -99,6 +99,7 @@ let MediatorService = class MediatorService {
                     connection,
                     recipientKey: (0, helpers_1.verkeyToDidKey)(message.to),
                     encryptedMessage: message.message,
+                    messageType: message.messageType,
                 });
         }
     }
